youtokentome/cpp/yttm.pyx

```diff
@@ -130,12 +132,30 @@ cdef class BPE:
             raise ValueError(status.message.decode())
         return subword.decode()
 
-    def decode(self, ids):
-        assert isinstance(ids, list)
+    def decode(self, ids, ignore_ids):
+
+        if not isinstance(ids, list):
+            raise TypeError(
+                "ids have to be a list instance but {} found".format(type(list))
+            )
+
+        if not (
+            isinstance(ignore_ids, Collection) or ignore_ids is None
+        ):
+            raise TypeError(
+                "ids have to be a Collection instance, but {} found".format(
+                    type(ignore_ids)
+                )
+            )
+
         if len(ids) > 0 and isinstance(ids[0], int):
             ids = [ids]
+        if ignore_ids is None:
+            ignore_ids = set([])
+        elif not isinstance(ignore_ids, set):
+            ignore_ids = set(ignore_ids)
```

Comment:
It is unnecessary. Just pass `set(ignore_ids)` without checking.

Reply:
Fixed